src/services/randamData.ts

```diff
@@ -0,0 +1,39 @@
+import axios from 'axios';
+
+import { tError } from '../model/contract';
+
+type tRandomResult = { results: Array<{
+    phone: string,
+    picture: {
+        large: string,
+        medium: string,
+        thumbnail: string
+    },
+    location: {
+      coordinates: {
+          latitude: string,
+          longitude: string
+      },
+    }
+  }> };
+
+export async function getRandomData(): Promise<{data: tRandomResult | null, error: tError | null}> {
+  try {
+    const { data } = await axios.get<tRandomResult>(
+      'https://randomuser.me/api/',
+      {
+        headers: {
+          Accept: 'application/json',
+        },
+      },
+    );
+    return { data, error: null };
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      console.error('error while getting random data. detail: ', error.message);
```

Comment:
Se for usar esse estilo de retornar o erro como parte do objeto retornado, talvez seja mais coerente retornar igual no Golang, com dois parâmetros err, data, por exemplo.

Reply:
O que mudaria nesse caso? seria trocar 'error' por 'err', ou em Javascript tem como retornar dois valores sem ser um objeto?

Reply:
Ele tem que retornar dois elementos "desestruturado" procura na internet.

Vamos fazer assim. Manda o texto do Tiago e daí eu penso como fazer

Reply:
Mandei lá